src/icons/__stories__/Icon.docs.mdx

```diff
@@ -0,0 +1,71 @@
+import { ArgsTable, Canvas, Story } from "@storybook/addon-docs"
+import Icon from "../Icon"
+
+# Icon
+
+Displays supported icon definitions from Font Awesome (Free by default, optionally Pro).
+
+<Canvas>
+  <Story id="icons-icon--icons" />
+</Canvas>
+
+## Props
+
+<ArgsTable of={Icon} />
+
+## Best Practices
+
+### When to Use
+
+- Use this component to add visual cues and enhance the user experience in interfaces such as buttons, navigation, and other interactive elements.
+
+### When Not to Use
+
+- Do not use this component for non-iconographic content, such as images or illustrations.
+
+### Usability
+
+- Use icons that are easily recognizable and have a clear meaning. Avoid using icons that are abstract or metaphorical.
+- Use the correct icon according to the context, meaning, and action associated with the element.
+
+### Content Guidelines
+
+- Use the Font Awesome icon set for standard icons.
+- Use the Pro icons option for additional icons that are not available in the free version.
+- For specific needs that are not covered by the Font Awesome icon set, use the CustomIcon component instead.
+
+### Accessibility Notes
+
+- Use the `aria-hidden` attribute for purely decorative icons that do not convey any information.
+- Provide a text alternative for icons that convey information, such as using an `aria-label` or `aria-labelledby` attribute.
+- Ensure that the icons are visible and can be easily distinguished by users with low vision or color blindness.
+- Avoid using only icons to convey information, as it can be difficult for users who rely on assistive technologies to understand the meaning of the icons.
+
+## Sizes
+
+By default, the icon will scale with container's font size.
+
+<Canvas>
+  <Story id="icons-icon--container-based-sizes" />
+</Canvas>
+
+You can use preset sizes to ensure consistency for specific use cases.
+
+<Canvas>
+  <Story id="icons-icon--preset-sizes" />
+</Canvas>
+
+## Theme Variables
+
+| Name                          | Description                | Default     |
+| ----------------------------- | -------------------------- | ----------- |
+| `--icon-utility-sm`           | sm Size                       | `0.75rem`   |
```

Comment:
I might be missing some context, but how did you decide on `utility-sm`, `utility-md`, `badge-lg`, and `badge-xl`? those names don't seem intuitive to me. Also there is a big jump in size between md and lg (1rem to 2.5rem), curious how those were decided as well